featuretests/cmd_juju_status_test.go

```diff
@@ -92,10 +92,147 @@ func (s *StatusSuite) TestMultipleRelationsInYamlFormat(c *gc.C) {
 }
 
 func (s *StatusSuite) TestMultipleRelationsInTabularFormat(c *gc.C) {
+	s.assertMultipleRelationsBetweenApplications(c)
 	context := s.run(c, "status", "--relations")
 	c.Assert(cmdtesting.Stdout(context), jc.Contains, `
 Relation provider      Requirer                   Interface  Type         Message
 wordpress:juju-info    logging:info               juju-info  subordinate  joining  
 wordpress:logging-dir  logging:logging-directory  logging    subordinate  joining  
 `[1:])
 }
+
+func (s *StatusSuite) assertServeralUnitsOnAMachines(c *gc.C) {
+	// Application A will have a unit on the same machine is a unit of an application B.
+	applicationA := s.Factory.MakeApplication(c, &factory.ApplicationParams{
+		Charm: s.Factory.MakeCharm(c, &factory.CharmParams{
+			Name:     "mysql",
+			Revision: "1",
+		}),
+	})
+
+	// Application B will have a unit on the same machine as a unit of an application A.
+	applicationB := s.Factory.MakeApplication(c, &factory.ApplicationParams{
+		Charm: s.Factory.MakeCharm(c, &factory.CharmParams{
+			Name:     "wordpress",
+			Revision: "3",
+		}),
+	})
+
+	// Put a unit from each, application A and B, on the same machine.
+	machine1 := s.Factory.MakeMachine(c, &factory.MachineParams{
+		Jobs:       []state.MachineJob{state.JobHostUnits},
+		InstanceId: instance.Id("id0"),
+	})
+	s.Factory.MakeUnit(c, &factory.UnitParams{
+		Application: applicationA,
+		Machine:     machine1,
+	})
+	s.Factory.MakeUnit(c, &factory.UnitParams{
+		Application: applicationB,
+		Machine:     machine1,
+	})
+
+}
+
+func (s *StatusSuite) TestStatusWhenFilteringByMachine(c *gc.C) {
+	s.assertServeralUnitsOnAMachines(c)
+
+	// Put a unit from the application C on a different machine.
```

Comment:
I didn't know what application C was until a dig in a bit. Maybe reword the comment just to say we are adding a new application to a machine ti ensure it is filtered out as expected.